### ISSUE TITLE
Go Doc should support pulling Raml documentation from source code

### ISSUE THREAD

**Issue by ObjectiveJoe** [golang/go#12460]
When using go code to generate rest api, it is beneficial to put them raml definitions of the api into the source code for the package implementing the rest api.

Issuing command: 'go doc -raml myservice' on a package with source code like this:

// RAML:
// baseUri: http://mydomain.github.io/myservice  
// title: MYSERVICE
// /projects:
//   /{project}:
//     /dependencies
//       /{dependency}: 

should produce output:

baseUri: http://mydomain.github.io/myservice
title: MYSERVICE
/projects:
  /{project}:
    /dependencies
      /{dependency}:

**Comment by minux** [comment]
why you can't just put the docs at an appropriate place
so that there is no need to change go doc (or godoc)
to display them?

**Comment by ObjectiveJoe** [comment]
I believe it promotes more up to date documentation to maintain the api specification with the source code; otherwise the code evolutes but the specification (used to generate the documentation) grows stale.

I do have a fix for this which I would like to get reviewed.  

Thanks

Joseph Hayes

**Comment by minux** [comment]
OK, let me rephrase my question:
should godoc (go doc) be extended to support all of the worlds
modeling languages?

If not, why is RAML special?

**Comment by ObjectiveJoe** [comment]
I thought of that. Notes almost work but it would be nice to have a variety that didn't require a uid and didn't echo the tag.

Joe

**Comment by ObjectiveJoe** [comment]
Also it's important the white spaces are left in tack.

If there was an optional tag called // RAW to start comments with would work in many situations.

Joe

**Comment by adg** [comment]
It's a non-goal of godoc to support comments that don't look like normal comments. We've worked hard toward this goal, and this proposal is a step backward. Sorry.

**Comment by ObjectiveJoe** [comment]
Andrew,

I do hear what you are saying.

Just for my own understanding, if you would be so kind, why is it that goal
of godoc was against having structured data in the comments? If you would
share the rational with me, I would sleep better tonight.

thanks,

Joe Hayes

**Comment by bradfitz** [comment]
We want comments that look like you're reading text, not reading a program.

We don't want to require a gibberish->HTML translator program to be able to read prose. The text as written should be readable directly.

**Comment by minux** [comment]
Go doc is for human to read, not for computers,
so there is no need to impose any syntactic structure.

That's also the reason we're not using @param and
annotations like that that are popular with other docs
systems.

see also http://blog.golang.org/godoc-documenting-go-code
